dietpi/dietpi-software

```diff
@@ -9878,6 +9922,42 @@ _EOF_
 
 		fi
 
+		software_id=186 # IPFS Node
+		if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == 1 )); then
+
+			Banner_Configuration
+
+			G_EXEC mkdir /mnt/dietpi_userdata/ipfs
+			Create_User -d /mnt/dietpi_userdata/ipfs ipfs
+			G_EXEC chown -R ipfs:ipfs /mnt/dietpi_userdata/ipfs
+
+			cat << '_EOF_' > /etc/systemd/system/ipfs.service
+[Unit]
+Description=IPFS (DietPi)
+Wants=network-online.target
+After=network-online.target dietpi-boot.service
+
+[Service]
+User=ipfs
+Environment=IPFS_PATH=/mnt/dietpi-userdata/ipfs
+ExecStart=/usr/local/bin/ipfs daemon
```

Comment:
Btw, not sure in this case, but generally in systemd units processes do not need to be daemonised, since the systemd service itself is. To allow system tracking the process state and catch STDOUT and STDERR for journalctl logging, it's best to run everything in foreground.

But in some cases foreground calls imply an interactive console and such things, which of course would be an overhead or even break the start, so probably it's not applicable here.

Reply:
I don't ever see the daemon forking into the background, it seems to just stay in the foreground.

Reply:
Indeed it stays in foreground, so all fine 👍.